docs/developers-guide.md

```diff
@@ -3,319 +3,130 @@
-export PATH=$PATH:$GOROOT/bin
-export user={your github profile name}
-export working_dir=$GOPATH/src/k8s.io
-```
-
-With your environment variables set up, clone the forks into your go environment.
-
-```
-mkdir -p $working_dir
-cd $working_dir
-git clone https://github.com/{user}/cloud-provider-openstack
-cd cloud-provider-openstack
-```
-
-If you want to build the plugins for single architecture:
-
-```
-# ARCH default to amd64
-export ARCH=arm64
-make build
-```
-
-And if you want to build the plugins for multiple architectures:
-
-```
-# ARCHS default to 'amd64 arm arm64 ppc64le s390x'
-export ARCHS='amd64 arm arm64 ppc64le s390x'
-make build-all-archs
-```
-
-If you want to run unit tests:
-
-```
-make test
-```
-
-#### Building inside container
-
-If you don't have a Go Environment setup, we also offer the ability to run make
-in a Docker Container.  The only requirement for this is that you have Docker
-installed and configured (of course).  You don't need to have a Golang
-environment setup, and you don't need to follow rules in terms of directory
-structure for the code checkout.
-
-To use this method, just call the `hack/make.sh` script with the desired argument:
-    `hack/make.sh build`  for example will run `make build` in a container.
-
-> NOTE: You MUST run the script from the root source directory as shown above,
-attempting to do something like `cd hack && make.sh build` will not work
-because we won't bind mount the source files into the container.
-
-### Getting and Building Kubernetes
+1. Install [Golang](https://golang.org/doc/install). You can choose the golang version same with the version defined in [go.mod](https://github.com/kubernetes/cloud-provider-openstack/blob/master/go.mod#L3) file.
+1. IDE. You can choose any IDE that supports Golang, e.g. Visual Studio Code, GoLand, etc.
+1. Install [Docker Engine](https://docs.docker.com/engine/install/) or other tools that could build container images such as [podman](https://podman.io/). After writing the code, the best way to test openstack-cloud-controller-manager is to replace the container image specified in its Deployment/StatefulSet/DaemonSet manifest in the kubernetes cluster. We are using docker in this guide.
 
-To get and build Kubernetes
+### Build openstack-cloud-controller-manager image
+In cloud-provider-openstack repo directory, run:
 
 ```
-cd $working_dir
-export KUBE_FASTBUILD=true
-git clone https://github.com/{user}/kubernetes
-cd kubernetes
-make cross
-```
-
-### Running the Cloud Provider
-
-To run the OpenStack provider, integrated with your cloud, be sure to have sourced the
-environment variables. You will also need to create an `/etc/kubernetes/cloud-config` file
-with the minimum options:
-
-```
-[Global]
-username=<username>
-password=<password>
-auth-url=http://<auth_endpoint>/v3
-tenant-id=<project_id>
-domain-id=<domain_id>
-```
-
-Start your cluster with the `hack/local-up-cluster.sh` with the proper environment variable set to
```

Comment:
For development purpose , isn't it better to keep this (hack/local-up-clustr.sh)
option as well to deploy cluster. I feel its easier to use when testing the PRs.

Reply:
`hack/local-up-clustr.sh` is the preferred way to install k8s for k8s development itself, and it's not even mentioned in k8s official installation doc. Additionally, there are lots of global environments used for the script that not even documented. So I removed that from the doc (and also replaced that with other approaches in some of our CI jobs)

Reply:
@lingxiankong right , we set a lot of variables which are not used also I guess :)
My thought is may be we could add link that this also one of the option to locally build and run k8s cluster as I have used it initially to run locally without issue easily. It might help new contributors to quickly run a cluster. Just the link ,https://github.com/kubernetes/community/blob/master/contributors/devel/running-locally.md#starting-the-cluster , if someone wants to explore , they could do that as well . wdyt?

Reply:
Agree with @ramineni that a link will definitely help people to dive in more in detail